JS install cleanup fixes

- Also ensure the minified JS is built before instaweb as it is referenced in
  the sed expression.

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar git-1.7.1.orig/Makefile git-1.7.1/Makefile
--- git-1.7.1.orig/Makefile	2010-04-24 02:38:35.000000000 +0000
+++ git-1.7.1/Makefile	2010-05-01 08:47:04.491916479 +0000
@@ -1561,17 +1561,16 @@
 	$(QUIET_SUBDIR0)gitweb $(QUIET_SUBDIR1) all
 
 ifdef JSMIN
-GITWEB_PROGRAMS += gitweb/gitweb.min.js
 GITWEB_JS = gitweb/gitweb.min.js
 else
 GITWEB_JS = gitweb/gitweb.js
 endif
 ifdef CSSMIN
-GITWEB_PROGRAMS += gitweb/gitweb.min.css
 GITWEB_CSS = gitweb/gitweb.min.css
 else
 GITWEB_CSS = gitweb/gitweb.css
 endif
+GITWEB_PROGRAMS += $(GITWEB_JS) $(GITWEB_CSS)
 OTHER_PROGRAMS +=  gitweb/gitweb.cgi  $(GITWEB_PROGRAMS)
 gitweb/gitweb.cgi: gitweb/gitweb.perl $(GITWEB_PROGRAMS)
 	$(QUIET_SUBDIR0)gitweb $(QUIET_SUBDIR1) $(patsubst gitweb/%,%,$@)
@@ -1586,7 +1585,7 @@
 endif # CSSMIN
 
 
-git-instaweb: git-instaweb.sh gitweb/gitweb.cgi gitweb/gitweb.css gitweb/gitweb.js
+git-instaweb: git-instaweb.sh gitweb/gitweb.cgi $(GITWEB_CSS) $(GITWEB_JS)
 	$(QUIET_GEN)$(RM) $@ $@+ && \
 	sed -e '1s|#!.*/sh|#!$(SHELL_PATH_SQ)|' \
 	    -e 's/@@GIT_VERSION@@/$(GIT_VERSION)/g' \
